README.md

```diff
@@ -473,6 +477,38 @@ of the contributors to perform the cherry-picks and carry-on testing of the olde
 The availability of stakeholder that can manage "service-oriented" maintenance and agrees to such a
 responsibility, will also drive our willingness to accept future, new providers to become community managed.
 
+### Suspending releases for providers
+
+In case a provider is found to require old dependencies that are not compatible with upcoming versions of
+the Apache Airflow or with newer dependencies required by other providers, the provider's release
+process can be suspended.
+
+This means:
+
+* The provider's status is set to "suspended"
+* No new releases of the provider will be made until the problem with dependencies is solved
+* Sources of the provider remain in the repository for now (in the future we might add process to remove them)
+* No new changes will be accepted for the provider (other than the ones that fix the dependencies)
+* The provider will be removed from the list of Apache Airflow extras in the next minor/major release
+  (2.7.0, 2.8.0, 3.0.0 etc.)
```

Comment:
Curious on the reasoning for excluding patch releases `#.#.1`  i.e. With the current example it means google-provider can't update until 2.6.0 which leaves the cases like the [google-ads](https://github.com/apache/airflow/pull/30067#issuecomment-1488521445) open for that much longer.

Reply:
@r-richmond. You are right. This is a good point. In this case we might also decide to make the suspension effective in the patch release.

Just to be clear - we are NOT going to suspend "google" provider. That's out of the question and Google is committed to upgrade all dependencies :). In fact the whole process here is to make it possible for them to upgrade, because currently it's `yandex` and `mysql` and `apache-beam` (on top of the dependency upgrades that google team does are holding back `google` provider from being upgraded. And the direction is clear:

The `yandex`, `mysql` and `beam` are restricting dependencies (protobuf <4 generally speaking) that make it impossible for `google` to upgrade their dependencies (generally to upgrade all clients that require protobuf >=4). So `google' provider need is the driving force for considering suspension of those three providers.

And strictly speaking, there is nothing "technology" side that would prevent us from doing so. Technically speaking it will happen automatically after we cherry-pick the "suspension" change to the 2-N-test branch. And we already agreed before and documented that **just** dependency in constraints (even breaking) change is not a breaking nor even significant  change for airflow and it happens all the time in the patchlevel releases. Note that users will still be able to (even if the constraints will be different) in most cases to install older provider versions together - for example even if new google provider introduces protobuf >=4, then users will **still** be able to install previous google provider + latest (suspended) yandes + latest (suspended) mysql : if we decide to suspend them. They just won't be able to upgrade to latest google if they decide to stay with yandex/mysql.

So yeah. I am in for making it even default to remove it from the next patchlevel release.